test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs

```diff
@@ -10118,6 +10107,80 @@ SimpleTestPackageContext createTestPackage(string name, string version, List<Sim
             }
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        [InlineData(null)]
+        public async Task RestoreNetCore_CPVMProject_ManagePackageVersionsCentrally_CanBeDisabled(bool? managePackageVersionsCentrally)
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+                var packagesForSource = new List<SimpleTestPackageContext>();
+                var packagesForProject = new List<SimpleTestPackageContext>();
+                var framework = FrameworkConstants.CommonFrameworks.NetCoreApp20;
+
+                SimpleTestPackageContext createTestPackage(string name, string version, List<SimpleTestPackageContext> source)
+                {
+                    var result = new SimpleTestPackageContext()
+                    {
+                        Id = name,
+                        Version = version
+                    };
+                    result.Files.Clear();
+                    source.Add(result);
+                    return result;
+                };
+
+                var projectA = SimpleTestProjectContext.CreateNETCore(
+                   "projectA",
+                   pathContext.SolutionRoot,
+                   framework);
+
+                // the package references defined in the project should not have version unless CPM is disabled
+                var packageBNoVersion = createTestPackage("B", managePackageVersionsCentrally is null || managePackageVersionsCentrally == true ? null : "1.0.0", packagesForProject);
+                var packageB100 = createTestPackage("B", "1.0.0", packagesForSource);
```

Comment:
It would be less for me to keep track of if this variable was either an explicit type and/or the variable name indicates it's a context.

```suggestion
                var packageContextB100 = createTestPackage("B", "1.0.0", packagesForSource);
```

Reply:
This is all just copy/paste so I didn't bother cleaning up poor patterns